pkg/controller/core/queuedworkload_controller.go

```diff
@@ -58,7 +63,29 @@ func NewQueuedWorkloadReconciler(queues *queue.Manager, cache *cache.Cache) *Que
 //+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queuedworkloads/finalizers,verbs=update
 
 func (r *QueuedWorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// No-op. All work is done in the event handler.
+	var wl kueue.QueuedWorkload
+	if err := r.client.Get(ctx, req.NamespacedName, &wl); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("queuedWorkload", klog.KObj(&wl))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling QueuedWorkload")
+
+	status := workloadStatus(&wl)
+	if status == pending && !r.queues.QueueExists(&wl) {
+		err := workload.UpdateWorkloadStatusIfChanged(ctx, r.client, &wl, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse,
+			fmt.Sprintf("Queue %s for workload doesn't exist", wl.Spec.QueueName), "Inadmissible")
+		return ctrl.Result{}, err
+	}
+	if wl.Spec.Admission != nil {
```

Comment:
this opposite, we check the condition if the workload is not yet admitted

Reply:
It can be pending if admitted by the scheduler but the clusterqueue doesn't exist, is the idea behind this check.